### PR TITLE
feat(ui/textfield): add email type option

### PR DESCRIPTION
**Pull request by jozefhruska**
Adds an "email" option for TextField's type. We need it on Cloud.

image

### CODE REVIEW

**Comment by jvorcak** [comment]
@jozefhruska can you please add more types as described in https://github.com/hazelcast/frontend-shared/issues/88 ? I think they'll be needed for billing section. 

Closes #88

File: packages/ui/src/TextField.tsx [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { Help, helpTooltipId } from './Help'
 
 import styles from './TextField.module.scss'
 
-type TextFieldTypes = 'number' | 'text' | 'password' | undefined
+type TextFieldTypes = 'email' | 'number' | 'password' | 'text' | undefined
```
Comment (aigoncharov) [inline packages/ui/src/TextField.tsx:15]: @jozefhruska could we have a story for the `email` field type?

**Comment by jozefhruska** [comment]
@jvorcak @aigoncharov Updated 👍

**Review by aigoncharov — APPROVED** [review]
LGTM!

**Pull request merged**